sklearn/preprocessing/data.py

```diff
@@ -2214,15 +2212,22 @@ def _transform_col(self, X_col, quantiles, inverse):
             upper_bound_x = 1
             lower_bound_y = quantiles[0]
             upper_bound_y = quantiles[-1]
-            #  for inverse transform, match a uniform PDF
+            #  for inverse transform, match a uniform distribution
             with np.errstate(invalid='ignore'):  # hide NaN comparison warnings
-                X_col = output_distribution.cdf(X_col)
+                if output_distribution == 'normal':
+                    X_col = stats.norm.cdf(X_col)
+                # else output distribution is already a uniform distribution
+
         # find index for lower and higher bounds
         with np.errstate(invalid='ignore'):  # hide NaN comparison warnings
-            lower_bounds_idx = (X_col - BOUNDS_THRESHOLD <
-                                lower_bound_x)
-            upper_bounds_idx = (X_col + BOUNDS_THRESHOLD >
-                                upper_bound_x)
+            if output_distribution == 'normal':
+                lower_bounds_idx = (X_col - BOUNDS_THRESHOLD <
+                                    lower_bound_x)
+                upper_bounds_idx = (X_col + BOUNDS_THRESHOLD >
+                                    upper_bound_x)
+            if output_distribution == 'uniform':
+                lower_bounds_idx = (X_col == lower_bound_x)
```

Comment:
should be ``lower_bounds_idx = (X_col < lower_bound_x)``?

Reply:
We need to use the equality `X_col == lower_bound_x` to properly map the lower and upper bounds (corresponding tests are failing if you use `<`). Also for `transform`, `lower_bound_x = min(X_col)`, therefore no index will satisfy `X_col < lower_bound_x`. Similar behavior for `inverse_transform`.

Reply:
Apologies my suggestion should be to something like this:
```
# find index for lower and higher bounds
with np.errstate(invalid='ignore'):  # hide NaN comparison warnings
    if output_distribution == 'normal':
        lower_bounds_idx = (X_col - BOUNDS_THRESHOLD <
                            lower_bound_x)
        upper_bounds_idx = (X_col + BOUNDS_THRESHOLD >
                            upper_bound_x)
        X_col[upper_bounds_idx] = upper_bound_y
        X_col[lower_bounds_idx] = lower_bound_y
```
I don't think we need this part when output_distribution == 'uniform'

Reply:
Output for one of the tests:
```python-traceback
________________________________________________________________________ test_quantile_transform_bounds ________________________________________________________________________

    def test_quantile_transform_bounds():
        # Lower and upper bounds are manually mapped. We checked that in the case
        # of a constant feature and binary feature, the bounds are properly mapped.
        X_dense = np.array([[0, 0],
                            [0, 0],
                            [1, 0]])
        X_sparse = sparse.csc_matrix(X_dense)

        # check sparse and dense are consistent
        X_trans = QuantileTransformer(n_quantiles=3,
                                      random_state=0).fit_transform(X_dense)
>       assert_array_almost_equal(X_trans, X_dense)
E       AssertionError:
E       Arrays are not almost equal to 6 decimals
E
E       (mismatch 83.33333333333333%)
E        x: array([[ 0.25,  0.5 ],
E              [ 0.25,  0.5 ],
E              [ 1.  ,  0.5 ]])
E        y: array([[0, 0],
E              [0, 0],
E              [1, 0]])
```
We still need to map the lower and upper bounds for the uniform distribution.

Reply:
Ahh, I see. So we need it because we use 
```
X_col[isfinite_mask] = .5 * (
    np.interp(X_col_finite, quantiles, self.references_)
    - np.interp(-X_col_finite, -quantiles[::-1],
                -self.references_[::-1]))
```
during transform, right?

Reply:
yes